add readme (#4)
Enapter/rockamalg: README.md

# Rockamalg

> Rockamalg amalgates lua files with all dependencies inside one lua file.

Reviewer: ```suggestion
Rockamalg amalgamates several Lua files with all their dependencies into a single Lua file.
```

> It uses [luarocks](https://luarocks.org/) to install dependencies and [amalg.lua](https://github.com/siffiejoe/lua-amalg/) to amalgate.

Reviewer: ```suggestion
It uses [LuaRocks](https://luarocks.org/) to install dependencies and [amalg.lua](https://github.com/siffiejoe/lua-amalg/) to amalgamate.
```

## Why?

> [Enapter blueprints](https://developers.enapter.com/docs/#blueprints) uses firmware as a single Lua file. But it can be complex and developer of firmware may want to split them in several modules or use existing modules. So we need a tool to combine multipple Lua modules into one single firmware file.

Reviewer: ```suggestion
[Enapter Blueprints](https://developers.enapter.com/docs/#blueprints) uses firmware as a single Lua file. But it can be complex to manage, and firmware developers may want to split them into several modules or use existing modules. So we need a tool to combine multiple Lua modules into a single firmware file.
```
